pkg/cluster/config/encoding/scheme.go

```diff
@@ -59,164 +52,39 @@ func AddToScheme(scheme *runtime.Scheme) {
+			return nil, err
 		}
 
-		// apply defaults
-		Scheme.Default(v1alpha1Cfg)
-
-		// converts to internal cfg. this will give Config with a single node with control-plane role
-		// NB. we are using custom conversion due to the complexity of conversion from v1alpha1 config
-		var internalCfg = &config.Config{}
-		v1alpha1Cfg.Convert(internalCfg)
-
-		return internalCfg, nil
-	}
-
-	// Otherwise it is a multi node, v1alpha2 config
-	var cfg = &config.Config{}
-
-	// Process all the yamlDocuments
-	for _, doc := range yamlDocuments {
-		// decode data into a Node object
-		var v1alpha2node = &v1alpha2.Node{}
-		_, _, err := Codecs.UniversalDecoder().Decode(doc.Contents, nil, v1alpha2node)
+		// decode data into a internal api Config object because
+		// to leverage on conversion functions for all the api versions
+		var cfg = &config.Config{}
+		err = runtime.DecodeInto(Codecs.UniversalDecoder(), contents, cfg)
 		if err != nil {
 			return nil, errors.Wrap(err, "decoding failure")
 		}
 
-		// apply defaults
-		Scheme.Default(v1alpha2node)
-
-		// converts to internal cfg
-		var internalNode = &config.Node{}
-		Scheme.Convert(v1alpha2node, internalNode, nil)
-
-		// adds the node to the internal config
-		// in case replicas shuold be generated for the nodes, Add will takes care.
-		if err := cfg.Add(internalNode); err != nil {
-			return nil, err
-		}
+		// converts back to the latest API version to apply defaults
+		Scheme.Convert(cfg, latestPublicConfig, nil)
 	}
 
-	return cfg, nil
-}
-
-// splitYAMLDocuments divides a yaml file into yamlDocuments,
-// and identifies each document with its own GroupVersionKind
-func splitYAMLDocuments(contents []byte) (yamlDocuments []yamlDocument, err error) {
-	yamlDocuments = []yamlDocument{}
-
-	// parses yamlDocuments separated by --- directives
-	buf := bytes.NewBuffer(contents)
-	reader := utilyaml.NewYAMLReader(bufio.NewReader(buf))
-	for {
-		// Read one YAML document at a time, until io.EOF is returned
-		documentContents, err := reader.Read()
-		if err == io.EOF {
-			break
-		} else if err != nil {
-			return nil, err
-		}
-		if len(documentContents) == 0 {
-			break
-		}
-
-		// Unmarshal the TypeMeta information of this byte slice
-		typeMetaInfo := runtime.TypeMeta{}
-		if err := yaml.Unmarshal(documentContents, &typeMetaInfo); err != nil {
-			return nil, err
-		}
-
-		// Require TypeMeta information to be present
-		if len(typeMetaInfo.APIVersion) == 0 || len(typeMetaInfo.Kind) == 0 {
-			return nil, errors.New("kind and apiVersion are mandatory information that needs to be specified in all YAML documents")
-		}
-
-		// Build a GroupVersionKind object from the TypeMeta object
-		documentGroupVersion, err := schema.ParseGroupVersion(typeMetaInfo.APIVersion)
-		if err != nil {
-			return nil, errors.Wrap(err, "unable to parse apiVersion")
-		}
-		documentGroupVersionKind := documentGroupVersion.WithKind(typeMetaInfo.Kind)
+	// apply defaults
+	Scheme.Default(latestPublicConfig)
```

Comment:
note to self: defaulting & loading should probably be distinct ... for combining with flags etc before defaulting ...